enterprise/admin/views.py

```diff
@@ -234,23 +238,38 @@ def _enroll_users(cls, enterprise_customer, emails, course_id, mode, request):
         non_existing = []
         failed = []
         enrollment_client = EnrollmentApiClient()
+        course_details = CourseApiClient().get_course_details(course_id)
+        course_url = request.build_absolute_uri(reverse('about_course', args=(course_id,)))
+        email_conn = mail.get_connection()
+        email_conn.open()
         for email in emails:
             try:
-                username = User.objects.get(email=email).username
+                user = User.objects.get(email=email)
             except User.DoesNotExist:
                 non_existing.append(email)
                 continue
             try:
-                enrollment_client.enroll_user_in_course(username, course_id, mode)
+                enrollment_client.enroll_user_in_course(user.username, course_id, mode)
             except HttpClientError as exc:
                 failed.append(email)
                 error_message = json.loads(exc.content.decode()).get("message", "No error message provided.")
                 logging.error(
                     "Error while enrolling user %(user)s: %(message)s",
-                    dict(user=username, message=error_message),
+                    dict(user=user.username, message=error_message),
                 )
             else:
```

Comment:
To my knowledge, they'd work the same at the moment, but a `continue` might cause issues later on if we want to add something that happens in the loop, even after an exception, later on. I chose not to modify the existing code.